java/src/hu/holyoil/controller/SunController.java

```diff
@@ -64,16 +65,18 @@ public void StartSunstorm()  {
         Queue<Asteroid> traversingQueue = new LinkedList<>();
 
         Random rand = new Random();
-        traversingQueue.add(
-                asteroids.get(
-                        rand.nextInt(asteroids.size())
-                )
-        );
+        Asteroid startingAsteroid = asteroids.get(rand.nextInt(asteroids.size()));
+        traversingQueue.add(startingAsteroid);
+        chosenAsteroids.add(startingAsteroid);
 
         while (chosenAsteroids.size() < asteroids.size() / 3 && !traversingQueue.isEmpty()) {
            Asteroid currentAsteroid = traversingQueue.remove();
-           chosenAsteroids.addAll(currentAsteroid.GetNeighbours());
-           traversingQueue.addAll(currentAsteroid.GetNeighbours());
+           List<Asteroid> untraversedNeighbours = currentAsteroid.GetNeighbours()
+                   .stream()
+                   .filter(chosenAsteroids::contains)
+                   .collect(Collectors.toList());
+           chosenAsteroids.addAll(untraversedNeighbours);
```

Comment:
This could potentially add more than 1/3 of all asteroids. For example if all asteroids are neighbours of starting asteroid

Reply:
On it ☕